hathor/merged_mining/coordinator.py

```diff
@@ -638,16 +638,17 @@ async def submit_to_hathor(self, job: SingleMinerJob, aux_pow: BitcoinAuxPow) ->
                 self.log.debug('share is too late, skip Hathor submit')
                 return
         try:
-            res = await self.coordinator.hathor_client.submit_block(block)
+            assert self.coordinator.hathor_mining is not None
+            res = await self.coordinator.hathor_mining.submit(block)
         except Exception:
             self.log.exception('submit to Hathor failed')
             return
-        self.log.debug('hathor_client.submit_block', res=res)
+        self.log.debug('hathor_mining.submit', res=res)
         if job.hathor_height is not None:
             self.coordinator.update_hathor_submitted(job.hathor_height)
         if res:
             self.log.info('Hurray!!! Hathor block found!!!', hash=block.hash.hex())
-            await self.coordinator.update_hathor_block()
+            # await self.coordinator.update_hathor_block()
```

Comment:
Why are we not updating the hathor block anymore here? Are we relying only on the periodic update?

Reply:
It's not periodic anymore, it's event-based, a new template will be sent as soon as the best block tip changes, making this obsolete. I will actually remove the commented line.